pkg/ec2helper/ec2helper.go

```diff
@@ -352,76 +353,137 @@ func (h *EC2Helper) getInstanceTypes(input *ec2.DescribeInstanceTypesInput) ([]*
 // Define all OS and corresponding AMI name formats
 var osDescs = map[string]map[string]string{
 	"Amazon Linux": {
-		"ebs":            "amzn-ami-hvm-????.??.?.????????.?-*-gp2",
-		"instance-store": "amzn-ami-hvm-????.??.?.????????.?-*-s3",
+		"ebs":            "amzn-ami-hvm-*",
+		"instance-store": "amzn-ami-hvm-*",
 	},
 	"Amazon Linux 2": {
-		"ebs": "amzn2-ami-hvm-2.?.????????.?-*-gp2",
+		"ebs": "amzn2-ami-hvm-*",
 	},
 	"Red Hat": {
-		"ebs": "RHEL-?.?.?_HVM-????????-*-?-Hourly2-GP2",
+		"ebs": "RHEL-9*",
 	},
 	"SUSE Linux": {
-		"ebs": "suse-sles-??-sp?-v????????-hvm-ssd-*",
+		"ebs": "suse-sles-*",
 	},
-	// Ubuntu 18.04 LTS
 	"Ubuntu": {
-		"ebs":            "ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-*-server-????????",
-		"instance-store": "ubuntu/images/hvm-instance/ubuntu-bionic-18.04-*-server-????????",
+		"ebs":            "ubuntu/images/*",
+		"instance-store": "ubuntu/images/*",
 	},
-	// 64 bit Microsoft Windows Server with Desktop Experience Locale English AMI
 	"Windows": {
-		"ebs": "Windows_Server-????-English-Full-Base-????.??.??",
+		"ebs": "Windows_Server-*-English-Full-Base*",
+	},
+}
+
+// Define all OS and corresponding AMI public parameters path in Parameter Store
```

Comment:
how are we determining the paths defined here? Did you validate the official image paths? Will they tend to change frequently? what happens to the ec2 if that happens we are just continuing the flow without any errors right, so do we expect to have old default images then?

Reply:
We follow the official doc: https://docs.aws.amazon.com/systems-manager/latest/userguide/parameter-store-finding-public-parameters.html which should not change frequently.

If the path is incorrect, will stop the flow. In our e2e_ec2helper_test, `TestGetLatestImages` and `TestGetImageIdsFromSSM` methods will protect and detect do we have correct paths.